### PR TITLE
CSI Migration Unit Tests

### PR DESCRIPTION
**Pull request by shanesiebken**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This is a WIP PR to get early feedback on added testing for CSI Migration.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by matthyx** [comment]
@shanesiebken is it still a WIP?

**Comment by shanesiebken** [comment]
@matthyx yeah, it is. I put this up for some feedback before the holidays, but getting it reviewed got lost in the shuffle. I'm going to be picking this back up and getting it done over the next few weeks.

File: pkg/controller/volume/attachdetach/attach_detach_controller_test.go [hunk 1]
```diff
@@ -136,6 +138,87 @@ func Test_AttachDetachControllerStateOfWolrdPopulators_Positive(t *testing.T) {
...
+	}
+
+	go adc.reconciler.Run(stopCh)
+	go adc.desiredStateOfWorldPopulator.Run(stopCh)
+	defer close(stopCh)
+
+	// Test that volumes in desiredStateOfWorld and actualStateOfWorld are migrated
+	csiNodesList, err := informerFactory.Storage().V1().CSINodes().Lister().List(labels.Everything())
+	for _, node := range csiNodesList {
+		attachedVolumes := adc.actualStateOfWorld.GetAttachedVolumesForNode(types.NodeName(node.Name))
```
Comment (davidz627) [inline pkg/controller/volume/attachdetach/attach_detach_controller_test.go]: where in this test are the actual volumes being submitted? It's not clear where the attached volumes are coming from and what their original types are, ideally it should be clear from reading the test from the top down "what workload/volume is being created (GCE?), assert this specific volume (check by name or UUID) is actually now a CSI volume".

Right now I don't see the creation, and I only see a blanket check that all volumes are `not CSI`. what if the volumes created at the start were not CSI to begin with?

File: pkg/controller/volume/attachdetach/attach_detach_controller_test.go [hunk 2]
```diff
@@ -136,6 +138,87 @@ func Test_AttachDetachControllerStateOfWolrdPopulators_Positive(t *testing.T) {
...
+		for _, volume := range attachedVolumes {
+			if volume.VolumeSpec == nil {
+				continue
+			}
+			if volume.VolumeSpec.Volume != nil {
+				if volume.VolumeSpec.Volume.CSI != nil {
```
Comment (davidz627) [inline pkg/controller/volume/attachdetach/attach_detach_controller_test.go]: doesn't `Volume.CSI != nil` imply that this is a CSI volume? Why is the error message that it `failed to translate to CSI`. That seems like the opposite of what this means

File: pkg/controller/volume/attachdetach/attach_detach_controller_test.go [hunk 3]
```diff
@@ -150,16 +233,90 @@ func Test_AttachDetachControllerRecovery(t *testing.T) {
...
 	podInformer := informerFactory.Core().V1().Pods().Informer()
-	var podsNum, extraPodsNum, nodesNum, i int
+	var extraPodsNum, nodesNum int
 
 	stopCh := make(chan struct{})
 
+	adc, plugins, _, nodesNum := testCaseSetup(t, fakeKubeClient, informerFactory, nodeInformer, csiNodeInformer, podInformer, stopCh)
+
+	// Populate ASW
+	err := adc.populateActualStateOfWorld()
```
Comment (davidz627) [inline pkg/controller/volume/attachdetach/attach_detach_controller_test.go]: why was this test case modified? the diff is pretty hard to read due to the extra changes being made around here. Could you split it out into 2 commits, one for the new test and one for miscellaneous changes? In general I would avoid making miscellaneous clean-up changes when trying to add significant new logic - makes it hard for reviewers to figure out what's "new"

**Review by davidz627 — COMMENTED** [review]
Thanks for working on this Shane! I think this is going in the right direction but right now the tests are a little bit hard to grok. A minor refactor could go a long way :)

**Pull request closed without merging**